File: Problems/0168_Excel_Sheet_Column_Title/Excel_Sheet_Column_Title.cs
using System;
using System.Collections.Generic;

public class Solution {
    public string ConvertToTitle(int n)
	{
        if (n < 1)
            return "";

        int temp = n - 1;
        var target = new List<int>();
		int mod, count = 0;

        while (true) {
            mod = temp % 26;
            target.Add(mod);
			count++;
            temp -= mod;
            if (temp >= 26) {
                temp = (int)(temp / 26) - 1;
                continue;
			}
            else
                break;
		}
            
        string result = "";

        for (int i = 0; i < count; ++i)
            result = (char)((int)'A' + target[i]) + result;
			Console.WriteLine("result = " + result);
		
        return result;
	}

	public void Main(string args)
	{
		string[] temp = args.Split('\t');

		System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
		sw.Start();
		
		Console.WriteLine("Result = " + ConvertToTitle(int.Parse(args)).ToString());
		
		sw.Stop();
		Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms");
	}
}

File: Problems/0168_Excel_Sheet_Column_Title/Excel_Sheet_Column_Title.py
# coding: utf-8

import sys
import time

class Solution:
    def convertToTitle(self, n):
        if n < 1:
            return ""

        temp = n - 1
        print("temp = %s" %temp)

        target = []
        while True:
            mod = temp % 26
            print("mod = %s" %mod)
            target.append(mod)
            temp -= mod
            if temp >= 26:
                temp = int(temp / 26) - 1
                continue
            else:
                break
            
        result = ""

        for i in range(len(target)):
            result = chr(ord('A') + target[i]) + result
            print("target[%s] = %s" %(i, target[i]))
            print("result = %s" %result)

        return result


def main():
    args = sys.argv
    argc = len(args)

    print("args[0] = %s %s" %(args[0], args[1]) )

    time0 = time.time()

    sl = Solution()
    print("result = %s" %(sl.convertToTitle(int(args[1]))))

    time1 = time.time()
    print("Execute time ... : %f[s]" %(time1 - time0))
    print()


if __name__ == "__main__":
    main()
